utils/tests/oauth-cp-requests.go

```diff
@@ -0,0 +1,88 @@
+package testutils
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/go-chi/chi/v5"
+)
+
+type CpResponseParams struct {
+	Timeout  time.Duration
+	Code     int
+	Response string
+}
+type CpResponseProducer struct {
+	Responses []CpResponseParams
+	CallCount int
+}
+
+func (s *CpResponseProducer) GetNext() CpResponseParams {
+	if s.CallCount >= len(s.Responses) {
+		panic("ran out of responses")
+	}
+	cpResp := s.Responses[s.CallCount]
+	s.CallCount++
+	return cpResp
+}
+
+func (cpRespProducer *CpResponseProducer) MockCpRequests() *chi.Mux {
+	srvMux := chi.NewMux()
+	srvMux.HandleFunc("/destination/workspaces/{workspaceId}/accounts/{accountId}/token", func(w http.ResponseWriter, req *http.Request) {
```

Comment:
Does this endpoints expect anything, or it expect POST?

Reply:
It expects only POST. Changes are done